app/src/main/java/com/babylon/wallet/android/presentation/rootdetection/RootDetectionDialog.kt

```diff
@@ -0,0 +1,64 @@
+package com.babylon.wallet.android.presentation.rootdetection
+
+import androidx.activity.compose.BackHandler
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.fillMaxSize
+import androidx.compose.foundation.layout.padding
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.LaunchedEffect
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.stringResource
+import com.babylon.wallet.android.MainActivity
+import com.babylon.wallet.android.R
+import com.babylon.wallet.android.designsystem.theme.RadixTheme
+import com.babylon.wallet.android.presentation.ui.composables.BasicPromptAlertDialog
+
+const val ROUTE_ROOT_DETECTION = "root_detection_route"
+
+@Composable
+fun RootDetectionContent(
+    viewModel: RootDetectionViewModel,
+    onAcknowledgeDeviceRooted: () -> Unit,
+    modifier: Modifier = Modifier
+) {
+    val activity = (LocalContext.current as MainActivity)
```

Comment:
wouldn't it be better to add parameter `onCloseApp: () -> Unit` instead of casting explicitly to MainActivity and making screen aware of implementation detail? We have this pattern already in the app, if you search for `onCloseApp` you will be able to find example usage

Reply:
> wouldn't it be better to add parameter `onCloseApp: () -> Unit` instead of casting explicitly to MainActivity and making screen aware of implementation detail? We have this pattern already in the app, if you search for `onCloseApp` you will be able to find example usage

Yup makes sense